src/lib/schema/feature-schema.test.ts

```diff
@@ -70,6 +70,56 @@ test('should allow weightType=fix', () => {
     expect(value).toEqual(toggle);
 });
 
+test('should not allow weightType=fix with floats', () => {
+    const toggle = {
+        name: 'app.name',
+        type: 'release',
+        project: 'default',
+        enabled: false,
+        impressionData: false,
+        stale: false,
+        archived: false,
+        strategies: [{ name: 'default' }],
+        variants: [
+            {
+                name: 'variant-a',
+                weight: 1.5,
+                weightType: 'fix',
+                stickiness: 'default',
+            },
+        ],
+    };
+
+    const { error } = featureSchema.validate(toggle);
+    expect(error.details[0].message).toEqual('Weight only supports 1 decimal');
```

Comment:
Why should this fail? There's only 1 decimal in the actual variant?

Reply:
The problem is that our weight range goes from 0 to 1000, where 1000 represents 100% and 1.5 represents 0.15%. This way of representing values as integers is common in representing money, but it is a bit confusing IMO, especially when you see things like this: 
1. I send 1 to the API (or 1.5 to the API)
2. I get 0.1% as result (I get only 1 decimal is supported when I sent only 1 decimal)

It has the overhead of having to know that the API payload is being interpreted differently

Reply:
Exactly, the error message is confusing but I wanted to test the two error messages (the old one in the text below and the new one in this test). The test description seems right IMO: `should not allow weightType=fix with floats` which is exactly what we want, but the message we're validating is a bit misleading from a backend perspective. If we could just validate against some error type (e.g. `INVALID_TYPE`) that'd be easier.